### ISSUE TITLE
Upload from CameraRoll to Firebase fails

### ISSUE THREAD

**Issue by MrRoyce** [wkh237/react-native-fetch-blob#109]
I've not been able to get any files successfully uploaded from the CameraRoll (on my Nexux 6 phone) to Firebase.  Essentially, the call to put the image to Firebase hangs.  It doesn't return the promise and nothing is logged to my console or to Firebase.

I've successfully run the sample app that loads images to Firebase, but that is using an image from https://avatars0.githubusercontent.com/u/5063785?v=3&s=460, not the CameraRoll.

Here is the code I'm using:

```javascript
if (newPhotos.length) {
  this.lastPhotoFetched = newPhotos[newPhotos.length - 1].uri;
  console.log('lastPhotoFetched', this.lastPhotoFetched);
  rnfbURI = RNFetchBlob.wrap(this.lastPhotoFetched);
  console.log('rnfbURI', rnfbURI);
  Blob
  .build(rnfbURI, { type : 'image/jpg;'})
  .then((blob) => {
    console.log('Image Name:', testImageName);
    console.log('blob:', blob);  // This is logged to the console
    // upload image using Firebase SDK
    firebaseApp.storage()  // This call never returns the promise
    .ref('images')
    .child(testImageName)
    .put(blob, { contentType : 'image/jpg' })
    .then((snapshot) => {
        console.log('Firebase Done.');  // This is not logged
        console.log(snapshot);
        blob.close();
    });
  })
  .catch((err) => {
      console.log('Blob err:', err);  // This is not logged
  });
}
```

Also, depending on which library I use, I get different values for the image.uri?   

If I use the library https://github.com/jeanpan/react-native-camera-roll-picker , the uri looks like `content://media/external/images/media/49793`

But if I use the https://github.com/bamlab/rn-camera-roll library, the uri looks like `file:/storage/emulated/0/DCIM/Camera/IMG_20160626_154134.jpg`

I'm not sure if that is an issue with the library, but it may be helpful in understanding the specific issue with Firebase

**Comment by smilechun** [comment]
Turn off debug remotely and try again.  For me,  it only works when debug remotely is off.
My spec:
RNv0.31
RN-blob 0.9.2 beta

**Comment by MrRoyce** [comment]
@smilechun, thanks for the tip.  I tried that and at least made some progress.  I am now getting an error:

*The callback slice() exists in module RNFFetchBlob, but only one callback may be registered to a function in a native module.*

So it looks like something is trying to happen.  Its tough to diagnose without being able to debug remotely.  Using console.warn at least allows me to show some information on the device, but its a tough slog!

**Comment by smilechun** [comment]
Try to use logcat,  it would helps

**Comment by wkh237** [comment]
@MrRoyce , it looks like the Blob has been created but FireBase does not successfully send request. You can turn on the logs of `XMLHttpRequest` and `Blob` by doing the following changes to `node_modules/react-native-fetch-blob/polyfill/Blob.js` and `node_modules/react-native-fetch-blob/polyfill/XMLHttpRequest.js`, the log might be a little overwhelm. 

```diff
- log.disable()
+ log.level(3)
```

Our fs module is able to consume the following URI 
- file path `/path/to/a/file.extension` (no need file:// prefix)
- Android media content database URI `content://media/path/uri`
- bundle assets `assets://asset/path`
- IOS assets library URI `assets-library://assets/library/uri`

According to your code snippet I think the usage is fine. If you want to make sure if the Blob has correct data, just print `_ref` or use `blob.blobPath`, that will return a file path which the blob data is stored. Then open or copy the file ..

For Android 

```
$ adb pull /path/to/the/blob/data /somewhere/of/your/pc/hdd
```

**Comment by smilechun** [comment]
The uri path should not start with "file: //" or "file: /",  keep the first "/",  for example "/xxxx/xxxx".  Please refer to the readme.md. it should stated what vaild uri it accepts.

**Comment by MrRoyce** [comment]
I was able to upload an image if I strip off the 'file:' prefix.  As I reported in the issues for the sample upload program, for some odd reason, Firebase wont let me re-upload that same image from my CameraRoll, it will fail after the 1st upload of that image, with a message saying the image cant be found.  Even though I can still see the image on my phone.

I guess I can close this issue, hopefully someone else can benefit from the findings.
